modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Details.tsx

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayLayout from '@clayui/layout';
+import classNames from 'classnames';
+import {fetch, navigate, openToast} from 'frontend-js-web';
+import React, {useRef, useState} from 'react';
+
+import {TFDSView} from '../FDSViews';
+import RequiredMark from '../RequiredMark';
+
+interface IDetailsProps {
+	fdsView: TFDSView;
+	fdsViewsAPIURL: string;
+	fdsViewsURL: string;
+	namespace: string;
+}
+
+const Details = ({
+	fdsView,
+	fdsViewsAPIURL,
+	fdsViewsURL,
+	namespace,
+}: IDetailsProps) => {
+	const [labelValidationError, setLabelValidationError] = useState(false);
+
+	const fdsViewDescriptionRef = useRef<HTMLInputElement>(null);
+	const fdsViewLabelRef = useRef<HTMLInputElement>(null);
+
+	const updateFDSView = async () => {
+		const body = {
+			description: fdsViewDescriptionRef.current?.value,
+			label: fdsViewLabelRef.current?.value,
+		};
+
+		const response = await fetch(`${fdsViewsAPIURL}/${fdsView.id}`, {
+			body: JSON.stringify(body),
+			headers: {
+				'Accept': 'application/json',
+				'Content-Type': 'application/json',
+			},
+			method: 'PATCH',
+		});
+
+		const responseJSON = await response.json();
+
+		if (responseJSON?.id) {
+			openToast({
+				message: Liferay.Language.get(
+					'your-request-completed-successfully'
+				),
+				type: 'success',
+			});
+
+			const controlMenuHeaderTitles = document.getElementsByClassName(
+				'control-menu-level-1-heading'
+			);
+
+			if (controlMenuHeaderTitles.length === 1) {
+				controlMenuHeaderTitles[0].innerHTML = Liferay.Util.escapeHTML(
+					fdsViewLabelRef.current?.value ?? ''
+				);
+			}
```

Comment:
IDK if we like DOM manipulation like this in React, I don't think they're really compatible 🤷